blocks/library/embed/index.js

```diff
@@ -110,20 +141,41 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 				const { url } = this.props.attributes;
 				const { setAttributes } = this.props;
 
+				// If we don't have any URL patterns, or we do and the URL doesn't match,
+				// then we should look for a block that has a matching URL pattern.
+				if ( ! patterns || ( patterns && ! matchesPatterns( url, patterns ) ) ) {
+					const matchingBlock = findBlock( url );
+					// WordPress blocks can work on multiple sites, and so don't have patterns.
+					if ( 'core-embed/wordpress' !== this.props.name && 'core/embed' !== matchingBlock ) {
+						// At this point, we have discovered a more suitable block for this url, so transform it.
+						if ( this.props.name !== matchingBlock ) {
+							this.props.onReplace( createBlock( matchingBlock, { url } ) );
+							return;
+						}
+					}
+				}
+
 				this.setState( { error: false, fetching: true } );
-				wp.apiRequest( { path: `/oembed/1.0/proxy?${ stringify( { url } ) }` } )
+				wpEmbedAPI( url )
 					.then(
 						( obj ) => {
 							if ( this.unmounting ) {
 								return;
 							}
-
-							const { html, provider_name: providerName } = obj;
```

Comment:
I'm confused. I thought the reason we needed to manually add all block regular expressions was because we had no means on the server to recognize the provider name. So where does this value originate?

Reply:
We have the means _sometimes_. The oEmbed API does provide this data, but in the case where the user has a plugin that handles YouTube embeds (for example) and bypasses the oEmbed code, we don't get any of the metadata, so we don't get the provider name. I guess we could handle this like `type`, and take it if it's there?

Reply:
So i'm going to remove this from the block settings, and use the block's title if we don't get a provider name from the embed API.